store.go

```diff
@@ -235,6 +236,73 @@ func (store *Store) Delete(ctx context.Context, prefix string, index int64) (ok
 	return
 }
 
+// Session return the session used to lock the given key. It returns an error
+// if the key is not locked or does not exist.
+func (store *Store) Session(ctx context.Context, key string) (sess Session, err error) {
+	var keyData KeyData
+	if keyData, err = store.readKeyData(ctx, key); err != nil {
+		return
+	}
+
+	if len(keyData.Session) == 0 {
+		err = fmt.Errorf("key %s is not locked", key)
+		return
+	}
+
+	var res io.ReadCloser
+	var configs []sessionConfig
+
+	path := "/v1/session/info/" + string(keyData.Session)
+	if _, res, err = store.client().do(ctx, "GET", path, nil, nil); err != nil {
+		return
+	}
+	defer res.Close()
+
+	dec := json.NewDecoder(res)
+	if err = dec.Decode(&configs); err != nil {
```

Comment:
How about using `store.client().Get(ctx, path, nil, &configs)` so we don't need to do all the JSON decoding in this function?